tools/tpm2_listpersistent.c

```diff
@@ -67,8 +68,9 @@ int readPublic(TSS2_SYS_CONTEXT *sapi_context,
     }
 
     printf("  {\n");
-    printf("\tType: 0x%x\n ", outPublic.t.publicArea.type);
-    printf("\tHash algorithm(nameAlg): 0x%x\n ", outPublic.t.publicArea.nameAlg);
+    printf("\tType: %s\n", tpm2_alg_util_algtostr(outPublic.t.publicArea.type));
+    printf("\tHash algorithm(nameAlg): %s\n",
+           tpm2_alg_util_algtostr(outPublic.t.publicArea.nameAlg));
```

Comment:
These need to be converted to tool output at some point.

Reply:
Actually we should keep the hex values and add the name to the output?

Hash Alg: sha256(0xB)

Reply:
@williamcroberts sounds good to me.

w.r.t tool output, I wondered about the same but thought what use could have this tool without any output. But now thinking about it, I think that everything should be converted to tool output besides the persistent handle.

That way someone can do something like this:

```
$ handle=$(tpm2_listpersistent -Q)
$ tpm2_evictcontrol -A o -H $handle -S $handle
```
What do you think?

Reply:
@williamcroberts Ok, I just pushed a commit to explain what I meant before reading this comment.

I thought that the idea of -Q was to be less verbose, I wonder what could be the case for someone just wanting the exit code of a tool that list stuff. But I see your point of the option being consistent.

Reply:
@martinezjavier thinkg of -q with grep, no output just lets you know if there is a match. Here it's less obvious of a use case (I don't have one per-say, outside of testing listpersistent command). But by redirecting stdout, you can control the output you need if its formatted nicely.

Reply:
If you want I can re-spin having the handle also printed with tpm2_tool_output().

In any case if someone finds that just printing the handle could be useful (I guess it may be like the example script snippet I shared), a tool specific option (-s, --summary ?) can be added to only prints the persistent handle.

Reply:
If the output is simply parsed via grep + cut, then we don't need any more support in the tool itself.

Reply:
@williamcroberts Ok, updated the last commit as you suggested.

Reply:
@williamcroberts agreed about just have parsed by other tools. Now what I think that could be useful though is being able to filter depending on the persistent object associated key and hash algorithms. Similar to what's done in tpm2_pcrlist.